aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1030,7 +1030,10 @@ static Tensor& norm_out(Tensor &result, const Tensor &self, const optional<Scala
 
   ScalarType out_dtype = result.defined() ? result.scalar_type() : (opt_dtype.has_value() ? opt_dtype.value() : toValueType(self.scalar_type()));
 
-  auto iter = make_reduction("norm", result, self, dim, keepdim, in_dtype, out_dtype);
+// omit in_dtype in the following call, to avoid make_reduction explicitly casting input to out_dtype
+  auto iter = isComplexType(self.scalar_type()) ?
+      make_reduction("vector_norm", result, self, dim, keepdim, in_dtype, out_dtype) :
```

Comment:
Why the name for complex is different? `vector_norm` vs `norm`

Reply:
No reason, just copy paste. I'll fix it, thanks for flagging.